src/Microsoft.Data.SqlClient/src/Resources/StringsHelper.cs

```diff
@@ -54,16 +55,69 @@ public static string GetString(string res, params object[] args)
                         args[i] = value.Substring(0, 1024 - 3) + "...";
                     }
                 }
-            #if NETFRAMEWORK
                 return string.Format(CultureInfo.CurrentCulture, res, args);
-            #else
-                return Format(res, args);
-            #endif
             }
             else
             {
                 return res;
             }
         }
+
+        // This method is used to decide if we need to append the exception message parameters to the message when calling Strings.Format. 
+        // by default it returns false.
+        // Native code generators can replace the value this returns based on user input at the time of native code generation.
+        // Marked as NoInlining because if this is used in an AoT compiled app that is not compiled into a single file, the user
+        // could compile each module with a different setting for this. We want to make sure there's a consistent behavior
+        // that doesn't depend on which native module this method got inlined into.
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool UsingResourceKeys()
+        {
+            return false;
+        }
+
+        public static string Format(string resourceFormat, params object[] args)
+        {
+            if (args is not null)
+            {
+                if (UsingResourceKeys())
+                {
+                    return resourceFormat + string.Join(", ", args);
+                }
+
+                return string.Format(resourceFormat, args);
+            }
+
+            return resourceFormat;
+        }
+
+        public static string Format(string resourceFormat, object p1)
+        {
+            if (UsingResourceKeys())
+            {
+                return string.Join(", ", resourceFormat, p1);
+            }
+
+            return string.Format(resourceFormat, p1);
+        }
+
+        public static string Format(string resourceFormat, object p1, object p2)
+        {
+            if (UsingResourceKeys())
+            {
+                return string.Join(", ", resourceFormat, p1, p2);
+            }
+
+            return string.Format(resourceFormat, p1, p2);
+        }
+
+        public static string Format(string resourceFormat, object p1, object p2, object p3)
+        {
+            if (UsingResourceKeys())
+            {
+                return string.Join(", ", resourceFormat, p1, p2, p3);
+            }
+
+            return string.Format(resourceFormat, p1, p2, p3);
+        }
```

Comment:
netcore specific methods.
```suggestion
#if !NETFRAMEWORK
        // This method is used to decide if we need to append the exception message parameters to the message when calling Strings.Format. 
        // by default it returns false.
        // Native code generators can replace the value this returns based on user input at the time of native code generation.
        // Marked as NoInlining because if this is used in an AoT compiled app that is not compiled into a single file, the user
        // could compile each module with a different setting for this. We want to make sure there's a consistent behavior
        // that doesn't depend on which native module this method got inlined into.
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static bool UsingResourceKeys()
        {
            return false;
        }

        public static string Format(string resourceFormat, params object[] args)
        {
            if (args is not null)
            {
                if (UsingResourceKeys())
                {
                    return resourceFormat + string.Join(", ", args);
                }

                return string.Format(resourceFormat, args);
            }

            return resourceFormat;
        }

        public static string Format(string resourceFormat, object p1)
        {
            if (UsingResourceKeys())
            {
                return string.Join(", ", resourceFormat, p1);
            }

            return string.Format(resourceFormat, p1);
        }

        public static string Format(string resourceFormat, object p1, object p2)
        {
            if (UsingResourceKeys())
            {
                return string.Join(", ", resourceFormat, p1, p2);
            }

            return string.Format(resourceFormat, p1, p2);
        }

        public static string Format(string resourceFormat, object p1, object p2, object p3)
        {
            if (UsingResourceKeys())
            {
                return string.Join(", ", resourceFormat, p1, p2, p3);
            }

            return string.Format(resourceFormat, p1, p2, p3);
        }
#endif
```

Reply:
Enclosed merged code from netcore with #if !NETFRAMEWORK.